h2o-algos/src/main/java/hex/generic/GenericModelOutput.java

```diff
@@ -39,7 +39,9 @@ public GenericModelOutput(final ModelDescriptor modelDescriptor, final ModelAttr
             _cross_validation_metrics_summary = convertTable(modelAttributes.getCrossValidationMetricsSummary());
 
             if (modelAttributes instanceof SharedTreeModelAttributes) {
-                fillSharedTreeModelAttributes((SharedTreeModelAttributes) modelAttributes, modelDescriptor);
+                _variable_importances = convertVariableImportances(((SharedTreeModelAttributes) modelAttributes).getVariableImportances());
```

Comment:
aah! now I see where is the difference. how about having
Optional<> extractVariableImportances(ModelAttributes a) { would contain the if - else if -else code }
and then
_variable_importances = extractVariableImportances(modelAttributes).orElseNull()
or even better, we could get make it even simple by making a interface of GenModelWithVariableImportances (or any such name) and make SharedTree and DL implement it)
or even betterer add an Optional<> getVariableImportnace() to the base ModelAtrributes?

Reply:
I like the first and third option. I don't like the interface option, as that would create an unnecessary pattern- there might be more in future. 

I guess third option it is. Thanks for suggesting that. It will make the code better.

Reply:
I will actually postpone this a little bit, as the backend could be all organized in that way. Not just the variable importance. Let's make it coherent in another PR to follow. 

There are some problems with that, e.g. the VariableImportance model is part of GenModel, which is JDK 7 and thus no optional, and the interfaces are problematic. And we actually do want distinct objects as an indicative of what is expected.

But I have two plans, One involves mapstruct and second one a general structure that only maps 1:1 at runtime from the JSON to the V3 objects we have in core.This will be a bigger change however.